### ISSUE TITLE
Improve AWS Macie integration rules

### ISSUE THREAD

**Issue by ace-109** [wazuh/wazuh#20623]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.7.0 | Ruleset | Manager/Agent | - | Ubuntu 22.04 |

Hello team,

I'm opening this issue because I observed the ruleset for classic AWS Macie doesn't work correctly for the new AWS Macie service. 

While working on https://github.com/wazuh/wazuh-documentation/issues/6601, I used the [sample macie findings](https://docs.aws.amazon.com/macie/latest/user/findings-samples.html) and also generated some macie events, the Wazuh rule description does not give enough information about what is going on in the alert

### Current result 
Find attached some alerts, the description is incomplete. 
<img width="955" alt="image" src="https://github.com/wazuh/wazuh/assets/41423652/6ddcf0b0-273c-4052-be20-7e3ec67e95e6">

While expanded, the fields are properly extracted
<img width="929" alt="image" src="https://github.com/wazuh/wazuh/assets/41423652/a9f80605-e313-47e7-afd0-f4bb7ba56c0b">

### Expected result
Accurate rule description with summary of alert, an example from cloudtrail is seem below
<img width="945" alt="image" src="https://github.com/wazuh/wazuh/assets/41423652/4ae90e4d-fee4-46b3-8115-e58f0f2f49f8">
 

I also attached the json copy of the Macie log.

<details>
<summary>json log sample</summary>

```
{
  "_index": "wazuh-alerts-4.x-2023.12.04",
  "_id": "n4pZNYwBjzgYpklPZxU1",
  "_version": 1,
  "_score": null,
  "_source": {
    "input": {
      "type": "log"
    },
    "agent": {
      "name": "wazuh-virtual-machine",
      "id": "000"
    },
    "manager": {
      "name": "wazuh-virtual-machine"
    },
    "data": {
      "integration": "aws",
      "aws": {
        "severity": {
          "score": "3",
          "description": "High"
        },
        "schemaVersion": "1.0",
        "log_info": {
          "s3bucket": "macie-my-ace",
          "log_file": "2023/12/04/14/PUT-S3-macie-1-2023-12-04-14-48-24-90000000-59fe-4dd3-aa7a-5d1e1fe18cbb"
        },
        "count": "2",
        "description": "All bucket-level block public access settings were disabled for the S3 bucket. Access to the bucket is controlled by account-level block public access settings, access control lists (ACLs), and the bucket’s bucket policy.",
        "source": "macie",
        "type": "Policy:IAMUser/S3BlockPublicAccessDisabled",
        "title": "Block public access settings are disabled for the S3 bucket",
        "sample": "false",
        "accountId": "10861290XXYY",
        "createdAt": "2023-12-04T14:14:21.317Z",
        "archived": "false",
        "partition": "aws",
        "id": "a1003b25-fc49-4343-bd8a-e7f090f40bd3",
        "policyDetails": {
          "actor": {
            "ipAddressDetails": {
              "ipOwner": {
                "asnOrg": "XMISSION",
                "org": "XMission, L.C.",
                "isp": "XMission, L.C.",
                "asn": "6315"
              },
              "ipCity": {
                "name": "Salt Lake City"
              },
              "ipAddressV4": "20Y.22X.146.253",
              "ipCountry": {
                "code": "US",
                "name": "United States"
              },
              "ipGeoLocation": {
                "lon": "-111.883000",
                "lat": "40.754500"
              }
            },
            "userIdentity": {
              "root": {
                "accountId": "10861290XXYY",
                "principalId": "10861290XXYY",
                "arn": "arn:aws:iam::108612904606:root"
              },
              "type": "Root"
            }
          },
          "action": {
            "actionType": "AWS_API_CALL",
            "apiCallDetails": {
              "lastSeen": "2023-12-04T14:32:12.000Z",
              "firstSeen": "2023-12-04T14:12:56.000Z",
              "apiServiceName": "s3.amazonaws.com",
              "api": "PutBucketPublicAccessBlock"
            }
          }
        },
        "region": "us-east-2",
        "category": "POLICY",
        "resourcesAffected": {
          "s3Bucket": {
            "owner": {
              "displayName": "null",
              "id": "ca006a0b4c389d85d6217b786a104c85ddeeedec50d90e34339567dc68c33434"
            },
            "createdAt": "2023-12-04T14:32:13.000Z",
            "publicAccess": {
              "effectivePermission": "NOT_PUBLIC",
              "permissionConfiguration": {
                "accountLevelPermissions": {
                  "blockPublicAccess": {
                    "blockPublicPolicy": "false",
                    "restrictPublicBuckets": "false",
                    "blockPublicAcls": "false",
                    "ignorePublicAcls": "false"
                  }
                },
                "bucketLevelPermissions": {
                  "accessControlList": {
                    "allowsPublicReadAccess": "false",
                    "allowsPublicWriteAccess": "false"
                  },
                  "bucketPolicy": {
                    "allowsPublicReadAccess": "false",
                    "allowsPublicWriteAccess": "false"
                  },
                  "blockPublicAccess": {
                    "blockPublicPolicy": "false",
                    "restrictPublicBuckets": "false",
                    "blockPublicAcls": "false",
                    "ignorePublicAcls": "false"
                  }
                }
              }
            },
            "name": "test-macie-myw",
            "defaultServerSideEncryption": {
              "encryptionType": "AES256"
            },
            "arn": "arn:aws:s3:::test-macie-myw",
            "tags": []
          }
        },
        "updatedAt": "2023-12-04T14:32:28.716Z"
      }
    },
    "rule": {
      "firedtimes": 1,
      "mail": false,
      "level": 8,
      "description": "AWS Macie High:  - .",
      "groups": [
        "amazon",
        "aws",
        "aws_macie"
      ],
      "id": "80354"
    },
    "location": "Wazuh-AWS",
    "decoder": {
      "name": "json"
    },
    "id": "1701702093.269951",
    "timestamp": "2023-12-04T18:01:33.080+0300"
  },
  "fields": {
    "data.aws.createdAt": [
      "2023-12-04T14:14:21.317Z"
    ],
    "data.aws.updatedAt": [
      "2023-12-04T14:32:28.716Z"
    ],
    "timestamp": [
      "2023-12-04T15:01:33.080Z"
    ]
  },
  "highlight": {
    "manager.name": [
      "@opensearch-dashboards-highlighted-field@wazuh-virtual-machine@/opensearch-dashboards-highlighted-field@"
    ],
    "rule.groups": [
      "@opensearch-dashboards-highlighted-field@amazon@/opensearch-dashboards-highlighted-field@"
    ]
  },
  "sort": [
    1701702093080
  ]
}
```

</details>

**Comment by fdalmaup** [comment]
# Issue Update

Found the root of the misbehavior but during the tests, I noticed that the events sent to Analysisd that contain `"severity": "CRITICAL"` do not trigger any alert although there is a rule for it:
https://github.com/wazuh/wazuh/blob/eb29da47d7253bedd10e34a281cccf221a5839eb/ruleset/rules/0350-amazon_rules.xml#L237-L243


When analyzing the sent event in depth, it was found that there is a difference in the hierarchy of the fields that causes the alert to not be triggered:
<details><summary>Critical event sent</summary>

```json
{
    "integration": "aws",
    "aws": {
        "log_info": {
            "aws_account_alias": "",
            "log_file": "2021/01/01/00/firehose_macie-1-2021-01-01-00-20-42-eb1f80bc-5d9d-426b-a0a1-e0ad8d3ef656",
            "s3bucket": "test-aws-wodle-macie"
        },
        "notification-type": "ALERT_CREATED",
        "tags": {
            "value": [
                "Open Permissions",
                "Basic Alert"
            ]
        },
        "name": "S3 Bucket IAM policy grants global read rights",
        "severity": "CRITICAL",
        "url": "https://mt.region.macie.aws.amazon.com/posts/arn%3Aaws%3Amacie%3Aregion%3AACCOUNT-ID%3Atrigger%2Fb731d9ffb1fe61508d4a478c92efa666%2Falert%2F8f8845fe3b28f25fb13b0d074044f3fc",
        "alert-arn": "arn:aws:macie:region:ACCOUNT-ID:trigger/b731d9ffb1fe61508d4a478c92efa666/alert/8f8845fe3b28f25fb13b0d074044f3fc",
        "risk-score": 9,
        "created-at": "2021-01-01T00:20:42.364509",
        "actor": "resources.test.com",
        "summary": {
            "Description": "S3 Bucket uses IAM policy to grant read rights to Everyone. Your IAM policy contains a clause that effectively grants read access to any user. Please audit this bucket, and data contained within and confirm that this is intentional. If intentional, please use the alert whitelist feature to prevent future alerts",
            "Bucket": [
                "resources.test.com"
            ],
            "Record Count": 1,
            "ACL": {
                "resources.test.com": {
                    "Owner": {
                        "DisplayName": "test",
                        "ID": "3ab1235e25ea9e94ff9b7e4e379ba6b0c872cd36c096e1ac8cce7df433b47101"
                    },
                    "Grants": [
                        {
                            "Grantee": {
                                "Type": "CanonicalUser",
                                "DisplayName": "test",
                                "ID": "3ab1235e25ea9e94ff9b7e4e379ba6b0c872cd36c096e1ac8cce7df433b47101"
                            },
                            "Permission": "FULL_CONTROL"
                        },
                        {
                            "Grantee": {
                                "Type": "Group",
                                "URI": "http://acs.amazonaws.com/groups/global/AllUsers"
                            },
                            "Permission": "READ"
                        }
                    ]
                }
            },
            "Event Count": 1,
            "Timestamps": [
                "2021-01-01T00:11:49.171020Z"
            ],
            "recipientAccountId": [
                "ACCOUNT-ID"
            ]
        },
        "source": "macie"
    }
}
```

</details>

This does not happen with some of the `High` severity ones that are found in `aws.severity.description` and matches with the default Macie rules:

<details><summary>High severity event</summary>

```json
{
    "integration": "aws",
    "aws": {
        "log_info": {
            "aws_account_alias": "",
            "log_file": "2021/05/04/00/firehose_macie-1-2021-12-01-00-40-40-e269348f-0b8e-4f61-a4e9-995424ftest1",
            "s3bucket": "wazuh-aws-wodle-macie"
        },
        "schemaVersion": "1.0",
        "id": "4fe51676-xxxxxxxxxxxx",
        "accountId": "5555564",
        "partition": "aws",
        "region": "us-east-1",
        "type": "Policy:IAMUser/S3BucketSharedEx",
        "title": "The S3 bucket is shared with an external AWS account.",
        "description": "The access control list (ACL) or bucket policy for the Amazon S3 bucket allows the bucket to be shared with an AWS account that isn\u2019t part of your Amazon Macie organization.",
        "severity": {
            "score": 3,
            "description": "High"
        },
        "createdAt": "2021-03-12T18:47:20.834Z",
        "updatedAt": "2021-04-22T21:10:17.764Z",
        "count": 59,
        "resourcesAffected": {
            "s3Bucket": {
                "arn": "arn:aws:s3:::em-xxxx-app-us-east-1-ss",
                "name": "em-xxxxxx-app-us-east-1-qa",
                "createdAt": "2018-04-05T23:53:47.000Z",
                "owner": {
                    "displayName": "em-aws-xxxx",
                    "id": "79xxxxxxxx"
                },
                "tags": [],
                "defaultServerSideEncryption": {
                    "encryptionType": "NONE"
                },
                "publicAccess": {
                    "permissionConfiguration": {
                        "bucketLevelPermissions": {
                            "accessControlList": {
                                "allowsPublicReadAccess": false,
                                "allowsPublicWriteAccess": false
                            },
                            "bucketPolicy": {
                                "allowsPublicReadAccess": true,
                                "allowsPublicWriteAccess": false
                            },
                            "blockPublicAccess": {
                                "ignorePublicAcls": false,
                                "restrictPublicBuckets": false,
                                "blockPublicAcls": false,
                                "blockPublicPolicy": false
                            }
                        },
                        "accountLevelPermissions": {
                            "blockPublicAccess": {
                                "ignorePublicAcls": false,
                                "restrictPublicBuckets": false,
                                "blockPublicAcls": false,
                                "blockPublicPolicy": false
                            }
                        }
                    },
                    "effectivePermission": "PUBLIC"
                },
                "allowsUnencryptedObjectUploads": "TRUE"
            }
        },
        "category": "POLICY",
        "policyDetails": {
            "action": {
                "actionType": "AWS_API_CALL",
                "apiCallDetails": {
                    "api": "PutBucketPolicy",
                    "apiServiceName": "s3.amazonaws.com",
                    "firstSeen": "2021-03-12T18:31:09.000Z",
                    "lastSeen": "2021-04-22T21:01:18.000Z"
                }
            },
            "actor": {
                "userIdentity": {
                    "type": "IAMUser",
                    "iamUser": {
                        "principalId": "AIxxxxxxxxxxx",
                        "arn": "arn:aws:iam::180000000:user/bots/svc_ccteam_jenkins",
                        "accountId": "180000000000",
                        "userName": "svc_cxxx"
                    }
                },
                "ipAddressDetails": {
                    "ipAddressV4": "144.5.12.0",
                    "ipOwner": {
                        "asn": "5000",
                        "asnOrg": "00",
                        "isp": "xxxxx",
                        "org": "xxxx"
                    },
                    "ipCountry": {
                        "code": "xx",
                        "name": "xxx xxx"
                    },
                    "ipCity": {
                        "name": "xxxx"
                    },
                    "ipGeoLocation": {
                        "lat": 0.0,
                        "lon": -0.0
                    }
                }
            }
        },
        "sample": false,
        "archived": false,
        "source": "macie"
    }
}
```

</details>

I will put this in common with the team before continuing with the necessary changes to the rules since we should consider the differences that may exist in the AWS Macie findings.

**Comment by fdalmaup** [comment]
# Issue Update

- In the [AWS Macie User Guide](https://docs.aws.amazon.com/macie/latest/user/findings-severity.html), it is stated that there are three severity levels (Low, Medium, and High) for the findings. There is no mention of a `CRITICAL` severity.
- The difference in the formats of the events in the bucket has already been reported in #8393, dated April 2021. The modifications of the events' format have been carried out at least three years ago.
- In the issue where the feature to fetch AWS Macie events was implemented (#1108), there are some related links now outdated that redirect to [this one](https://docs.aws.amazon.com/macie/latest/userguide/what-is-macie.html), in which it is explained that  `AWS Macie Classic` is discontinued. 
- Viewing the previously mentioned links in https://web.archive.org/ it was found that the events generated by AWS Macie were considered alerts and included the `Critical` severity level:
![image](https://github.com/wazuh/wazuh/assets/26691145/46d5857a-0cc7-4919-8b80-173ea19e1156)
The new AWS Macie does not generate alerts but findings instead.

In conclusion, it is safe to ignore and stop supporting the old format given its age. A new analysis of the AWS Macie findings, removal of the old alerts, and generation of new events will be carried out in #20697.
